paddle/operators/recurrent_op.h

```diff
@@ -117,10 +130,14 @@ class RecurrentOp final : public framework::OperatorBase {
     alg_.Run(scope, dev_ctx);
   }
 
+  void set_stepnet(std::shared_ptr<NetOp> net) { stepnet_ = net; }
+  const NetOp* stepnet() const { return stepnet_.get(); }
+
   static const rnn::ArgumentName kArgName;
 
  private:
   RecurrentAlgorithm alg_;
+  std::shared_ptr<NetOp> stepnet_;
```

Comment:
just like `NetOp`, RNNOp has its own python binding to expose some method, such as `set_stepnet`.

In `pybind.cc`, all the operators created by user are handled by `shared_ptr`, maybe easier to be referenced across different levels, when user create a `Operator`, a shared_ptr is returned, it is free to use the pointer, or pass the pointer into multiple `NetOp` (add an op into a net).

Operators will live until the whole execution period,  so a `shared_ptr` seems reasonable.